[Metricbeat] VSphere module dashboards

philippkahr opened the pull request:
Hi,

I created some dashboards for the VSphere module.

https://github.com/elastic/beats/issues/14101

elasticmachine commented:
Since this is a community submitted pull request, a Jenkins build has not been kicked off automatically. Can an Elastic organization member please verify the contents of this patch and then kick off a build manually?

elasticmachine commented:
Since this is a community submitted pull request, a Jenkins build has not been kicked off automatically. Can an Elastic organization member please verify the contents of this patch and then kick off a build manually?

philippkahr commented:
The Travis build is failing with 

```text
Get the licenses available from ['./vendor', './metricbeat/vendor', './metricbeat/module/kvm/vendor', './metricbeat/module/postgresql/vendor', './metricbeat/module/vsphere/vendor', './metricbeat/module/mysql/vendor']
```

but I do not know how to fix this.

ChrsMark commented on metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-host.json:
```diff
@@ -0,0 +1,1286 @@
+              }
+            ]
+          },
+          "title": "Free vs Used CPU on Cluster stacked [metricbeat VSphere] ECS",
```
```suggestion
          "title": "Free vs Used CPU on Cluster stacked [Metricbeat VSphere] ECS",
```

ChrsMark commented on metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-host.json:
```diff
@@ -0,0 +1,1286 @@
+            }
+          }
+        },
+        "title": "top datastore used [Metricbeat VSphere] ECS",
```
```suggestion
        "title": "Top datastore used [Metricbeat VSphere] ECS",
```

ChrsMark commented on metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json:
```diff
@@ -0,0 +1,850 @@
+          }
+        },
+        "savedSearchRefName": "search_0",
+        "title": "how many VMs run on a ESXI host [Metricbeat VSphere] ECS",
```
```suggestion
        "title": "VMs per ESXI host [Metricbeat VSphere] ECS",
```

ChrsMark commented on metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json:
```diff
@@ -0,0 +1,850 @@
+          }
+        },
+        "savedSearchRefName": "search_0",
+        "title": "OS Pie [Metricbeat VSphere] ECS",
```
```suggestion
        "title": "OS Distribution [Metricbeat VSphere] ECS",
```

philippkahr commented:
Hi @ChrsMark 

thanks for reviewing my changes. Sorry for the "typos", maybe we should add a naming scheme and example to the contributions guide?

> 1. Could you add an updated screenshot of the dashboards in this PR so as people can comment on the visualizations and propose improvements to make them more meaningful?

Sure! I will do that later tonight.

> 2. I wonder if it is ok to have 2 different dashboards for this module.
Hmm, that depends really. I think the first dashboard is really for the cluster and hosts. The other one is a drill down into the stats of the virtual machine. Combining those would not make much sense.

ChrsMark commented:
> Hi @ChrsMark
> 
> thanks for reviewing my changes. Sorry for the "typos", maybe we should add a naming scheme and example to the contributions guide?
> 

Sure! We are currently in discussions about improving the Developer Guides in general. Feel free to open an issue mentioning that proposal.

ChrsMark commented:
Thank you for taking the time to open the issue @philippkahr!

We can go on with this PR regardless, so feel free to add screenshot(s) of the dashboards in new comments here so as to verify them and maybe suggest improvements. 🙂

Jarzamendia commented:
I'm very excited to see this in action! My team are very happy with the Infrastructure dashboard. If we can also see the VSphere metrics to, this stack will add so much value to our environment!

philippkahr commented:
Hi,

sorry, it took so long.

I had to add a lot of white boxes and some example text since I am using real data. I hope that this is still ok.

## Metricbeat VSphere  Hosts Overview ECS
![Metricbeat VSphere  Hosts Overview ECS](https://user-images.githubusercontent.com/12175559/67968314-70dccd00-fc07-11e9-9686-645e408baede.png)

## Metricbeat VSphere  VMs overview ECS
![Metricbeat VSphere  VMs overview ECS](https://user-images.githubusercontent.com/12175559/67968311-6de1dc80-fc07-11e9-9f95-262f1100d670.png)

The heatmap itself is used to show where a VM has been on an ESXi host in the timeframe. This might show you if a VM is likely to be moved around a lot. The other dashboards are self-explanatory, to me at least :).

Jarzamendia commented:
A very good view would be a VM distributed by the hosts. Similar to what you see on the Infrastructure tab. This would allow us to quickly identify hosts that could soon become overloaded if we know which VMs will soon have access spikes.

ChrsMark commented on metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json:
```diff
@@ -0,0 +1,850 @@
+          }
+        },
+        "savedSearchRefName": "search_0",
+        "title": "VM running on what ESXi host [Metricbeat VSphere] ECS",
```
```suggestion
        "title": "VMs' placements on ESXi Hosts[Metricbeat VSphere] ECS",
```

ChrsMark commented on metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json:
```diff
@@ -0,0 +1,850 @@
+              }
+            ]
+          },
+          "title": "VM running on what ESXi host [Metricbeat VSphere] ECS",
```
```suggestion
          "title": "VM placements on ESXi Hosts[Metricbeat VSphere] ECS",
```

ChrsMark commented on metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json:
```diff
@@ -0,0 +1,850 @@
+          }
+        },
+        "savedSearchRefName": "search_0",
+        "title": "networks from virtualmachine [Metricbeat VSphere] ECS",
```
Not so sure what this chart shows but maybe we could change `virtualmachine` to `VM`?
```suggestion
        "title": "Networks from virtualmachine [Metricbeat VSphere] ECS",
```

ChrsMark commented on metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json:
```diff
@@ -0,0 +1,850 @@
+            "legendPosition": "right",
+            "type": "pie"
+          },
+          "title": "networks from virtualmachine [Metricbeat VSphere] ECS",
```
```suggestion
          "title": "Networks from virtualmachine [Metricbeat VSphere] ECS",
```

ChrsMark commented on metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json:
```diff
@@ -0,0 +1,850 @@
+            },
+            "totalFunc": "sum"
+          },
+          "title": "virtualmachine used CPU total ram network [Metricbeat VSphere] ECS",
```
```suggestion
          "title": "CPU/RAM/Network per VM [Metricbeat VSphere] ECS",
```

ChrsMark commented on metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-host.json:
```diff
@@ -0,0 +1,1286 @@
+            }
+          }
+        },
+        "title": "Top datastore used [Metricbeat VSphere] ECS",
```
```suggestion
        "title": "Top Datastore Used [Metricbeat VSphere] ECS",
```

ChrsMark commented on metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-host.json:
```diff
@@ -0,0 +1,1286 @@
+            "time_field": "",
+            "type": "gauge"
+          },
+          "title": "top datastore used [Metricbeat VSphere] ECS",
```
```suggestion
          "title": "Top Datastore Used [Metricbeat VSphere] ECS",
```

ChrsMark commented on metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-host.json:
```diff
@@ -0,0 +1,1286 @@
+          }
+        },
+        "savedSearchRefName": "search_0",
+        "title": "Datastore types [Metricbeat VSphere] ECS",
```
```suggestion
        "title": "Datastore Types [Metricbeat VSphere] ECS",
```

ChrsMark commented:
Hi @philippkahr ! Thank you for adding the screenshots. :)

In general it looks good to me! I just added some minor suggestions so as to keep the namings consistent.

ChrsMark commented:
> A very good view would be a VM distributed by the hosts. Similar to what you see on the Infrastructure tab. This would allow us to quickly identify hosts that could soon become overloaded if we know which VMs will soon have access spikes.

I think that the Heatmap @philippkahr adds can give this information, since it actually shows the placements of the VMs on each Host.

kaiyan-sheng commented:
Sorry for the late review. This dashboard looks great! Could you please also document it in `beats/metricbeat/module/vsphere/_meta/docs.asciidoc`? Something like: 
```
[float]
=== Dashboard

The vsphere module comes with a predefined dashboard. For example:

image::./images/metricbeat_vsphere_dashboard.png[]
```
This will also require to put a screenshot of this dashboard `metricbeat_vsphere_dashboard.png` into `beats/metricbeat/docs/images`. Then it will show up in documentation for vsphere module 😄

philippkahr commented:
@ChrsMark thanks for the suggestions

@kaiyan-sheng I've added the information as requested :)

ChrsMark commented:
@philippkahr great! you will need again `make update` since you changed the docs. Also if didn't already, please rebase on top of the latest master.

philippkahr commented:
Hi @ChrsMark 

can you please double-check if I did the rebase correctly? I am doing this for the first time. 

```
inside the vspheredashboard branch from my forked repo
$ git remote -v

origin  git@github.com:philippkahr/beats.git (fetch)
origin  git@github.com:philippkahr/beats.git (push)
upstream        https://github.com/elastic/beats.git (fetch)
upstream        https://github.com/elastic/beats.git (push) 

$ git rebase upstream/master

First, rewinding head to replay your work on top of it...
Applying: https://github.com/elastic/beats/issues/14101 renamed everything according to xyz [metricbeat] ECS, imported and exported the dashboards, created the module.yml
Applying: https://github.com/elastic/beats/pull/14135 added dashboard
Applying: https://github.com/elastic/beats/pull/14135 trying to fix the error from travis ci
Applying: https://github.com/elastic/beats/pull/14135 fixing naming pattern missmatch
Applying: https://github.com/elastic/beats/pull/14135: fixed: empty description, fixed: expected title with format 'Some title [Metricbeat Module]'
Applying: https://github.com/elastic/beats/pull/14135: travis told me to run `Error: some files are not up-to-date. Run 'mage fmt update' then review and commit the changes. Modified: [metricbeat/docs/modules_list.asciidoc]`
Applying: https://github.com/elastic/beats/pull/14135 Remove submodule beats from vendor
Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json
Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json
Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-host.json
Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-host.json
Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json
Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json
Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json
Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-host.json
Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-host.json
Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-host.json
Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json
Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json
Applying: https://github.com/elastic/beats/pull/14135 added the vsphere dashboards to the docs
Applying: make update

$ git push -f origin vspheredashboards 

Enumerating objects: 211, done.
Counting objects: 100% (211/211), done.
Delta compression using up to 8 threads
Compressing objects: 100% (161/161), done.
Writing objects: 100% (197/197), 1.78 MiB | 2.33 MiB/s, done.
Total 197 (delta 138), reused 10 (delta 3)
remote: Resolving deltas: 100% (138/138), completed with 13 local objects.
To github.com:philippkahr/beats.git
 + 13de30c44...41c36d5a3 vspheredashboards -> vspheredashboards (forced update)

```

ChrsMark commented:
> Hi @ChrsMark
> 
> can you please double-check if I did the rebase correctly? I am doing this for the first time.
> 
> ```
> inside the vspheredashboard branch from my forked repo
> $ git remote -v
> 
> origin  git@github.com:philippkahr/beats.git (fetch)
> origin  git@github.com:philippkahr/beats.git (push)
> upstream        https://github.com/elastic/beats.git (fetch)
> upstream        https://github.com/elastic/beats.git (push) 
> 
> $ git rebase upstream/master
> 
> First, rewinding head to replay your work on top of it...
> Applying: https://github.com/elastic/beats/issues/14101 renamed everything according to xyz [metricbeat] ECS, imported and exported the dashboards, created the module.yml
> Applying: https://github.com/elastic/beats/pull/14135 added dashboard
> Applying: https://github.com/elastic/beats/pull/14135 trying to fix the error from travis ci
> Applying: https://github.com/elastic/beats/pull/14135 fixing naming pattern missmatch
> Applying: https://github.com/elastic/beats/pull/14135: fixed: empty description, fixed: expected title with format 'Some title [Metricbeat Module]'
> Applying: https://github.com/elastic/beats/pull/14135: travis told me to run `Error: some files are not up-to-date. Run 'mage fmt update' then review and commit the changes. Modified: [metricbeat/docs/modules_list.asciidoc]`
> Applying: https://github.com/elastic/beats/pull/14135 Remove submodule beats from vendor
> Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json
> Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json
> Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-host.json
> Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-host.json
> Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json
> Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json
> Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json
> Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-host.json
> Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-host.json
> Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-host.json
> Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json
> Applying: Update metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json
> Applying: https://github.com/elastic/beats/pull/14135 added the vsphere dashboards to the docs
> Applying: make update
> 
> $ git push -f origin vspheredashboards 
> 
> Enumerating objects: 211, done.
> Counting objects: 100% (211/211), done.
> Delta compression using up to 8 threads
> Compressing objects: 100% (161/161), done.
> Writing objects: 100% (197/197), 1.78 MiB | 2.33 MiB/s, done.
> Total 197 (delta 138), reused 10 (delta 3)
> remote: Resolving deltas: 100% (138/138), completed with 13 local objects.
> To github.com:philippkahr/beats.git
>  + 13de30c44...41c36d5a3 vspheredashboards -> vspheredashboards (forced update)
> ```

Yeah it looks ok, this is the way to go with rebasing. Thank you!

ChrsMark commented:
ok to test

ChrsMark commented on metricbeat/module/vsphere/_meta/kibana/7/dashboard/metricbeat-vsphere-vm.json:
```diff
@@ -0,0 +1,850 @@
+          }
+        },
+        "savedSearchRefName": "search_0",
+        "title": "virtualmachine used CPU total ram network [Metricbeat VSphere] ECS",
```
```suggestion
        "title": "CPU/RAM/Network per VM [Metricbeat VSphere] ECS",
```

ChrsMark commented:
Loaded the dashboards successfully. Found one left-over on one of the tittles.

After that I think we are ok to merge.

ChrsMark commented:
ok to test

ChrsMark commented:
Failing tests, seem to be irrelevant. Merging.

Thank you so much for contributing this @philippkahr 🙂 !
